### ISSUE TITLE
Building tag 5.0.10 fails installing dotnet

### ISSUE THREAD

**Issue by ta264** [dotnet/aspnetcore#36834]
### Describe the bug
Trying to build aspnetcore from the 5.0.10 tag fails with the following error:

```
dotnet-install: Downloading primary link https://dotnetcli.azureedge.net/dotnet/Runtime/5.0.10-servicing.21403.8/dotnet-runtime-5.0.10-servicing.21403.8-linux-x64.tar.gz
curl: (22) The requested URL returned error: 404 
dotnet-install: The resource at https://dotnetcli.azureedge.net/dotnet/Runtime/5.0.10-servicing.21403.8/dotnet-runtime-5.0.10-servicing.21403.8-linux-x64.tar.gz is not available.
dotnet_install: Error: Could not find `.NET Core Runtime` with version = 5.0.10-servicing.21403.8
dotnet_install: Error: Refer to: https://aka.ms/dotnet-os-lifecycle for information on .NET Core support
Failed to install dotnet SDK from public location (exit code '1').
Failed to install dotnet SDK from public location (exit code '1').
/home/vsts/work/1/s/aspnetcore/.packages/microsoft.dotnet.arcade.sdk/5.0.0-beta.21328.3/tools/InstallDotNetCore.targets(15,5): error : dotnet-install failed [/home/vsts/work/1/s/aspnetcore/.packages/microsoft.dotnet.arcade.sdk/5.0.0-beta.21328.3/tools/Tools.proj]
##[error].packages/microsoft.dotnet.arcade.sdk/5.0.0-beta.21328.3/tools/InstallDotNetCore.targets(15,5): error : (NETCORE_ENGINEERING_TELEMETRY=Restore) dotnet-install failed
dotnet-install: Note that the intended use of this script is for Continuous Integration (CI) scenarios, where:
dotnet-install: - The SDK needs to be installed without user interaction and without admin rights.
dotnet-install: - The SDK installation doesn't need to persist across multiple CI runs.
dotnet-install: To set up a development environment or to run apps, use installers rather than this script. Visit https://dotnet.microsoft.com/download to get the installer.
```

### To Reproduce
- Checkout `v5.0.10` from the repository
- Run `./build.sh -c Release -ci -pack -nobl /p:CrossgenOutput=false /p:OfficialBuildId=$(date +%Y%m%d)-99`

### Further technical details
Perhaps related to https://github.com/dotnet/runtime/issues/59250 If the release tag process failed for runtime perhaps it didn't upload the required artifacts also?

**Comment by javiercn** [comment]
@ta264 thanks for contacting us.

@dotnet/aspnet-build can you take a look?

**Comment by dougbu** [comment]
The `v5.0.10` tag can't build for us either; the `5.0.10-servicing.21403.8` runtime version was never made publicly available. 

Fix is in the 51a999890f85 commit. The other option is to checkout the 'release/5.0' branch instead.

/cc @mmitche

**Comment by dougbu** [comment]
Thank you for your feedback. We're closing this issue as the questions asked here have been answered.
